game/world/objects/npc.cpp

```diff
@@ -1354,9 +1354,24 @@ bool Npc::implGoTo(uint64_t dt, float destDist) {
   else if(mvAlgo.isClose(go2.target(),destDist)) {
     bool finished = true;
     if(go2.flag==GT_Way) {
+      if(std::abs(go2.target().y-y)>destDist+100) {
+        recalculateWayPath();
+        mvAlgo.tick(dt);
+        return true;
+        }
       go2.wp = wayPath.pop();
       if(go2.wp!=nullptr) {
         attachToPoint(go2.wp);
+        if(std::abs(go2.wp->x-x)+std::abs(go2.wp->z-z)<std::abs(go2.wp->y-y) && !isMonster()) {
```

Comment:
> std::abs(go2.wp->x-x)+std::abs(go2.wp->z-z)<std::abs(go2.wp->y-y)

This condition is supper random. 

> isMonster

Why monster? This condition will ban skeletons, for example

Reply:
> This condition is supper random.

Checks if angle in y-direction is greater than 45 degrees. This supposed to be a filter for ladder that prevents `availableMob` form being called every time if there's no ladder. But this point in code is not reached that often, so could also leave it out 

> Why monster? This condition will ban skeletons, for example
These waypoints below the map pass the 45 degree check and are mostly used by monsters. Wasn't aware of the fact that skeletons can use ladders.

Reply:
> 45 degree check

This math is not correct, regards checking for 45-degree. More like 45-60 range, based on x-to-z ratio.